src/pudl/constants.py

```diff
@@ -2678,7 +2689,7 @@
         'owner_utility_id_eia': pd.Int64Dtype(),
         'owner_zip_code': pd.StringDtype(),  # Must preserve leading zeroes.
         # we should transition these into readable codes, not a one letter thing
-        'ownership_code': pd.StringDtype(),
+        'ownership_code': pd.StringDtype(),  # Used by AES for Merger table
```

Comment:
Does this column take on all the same values that appear in the other tables that have it? I think we were going to make this into an enumerated set of categories eventually, but weren't sure what all the potential values are.

Reply:
I couldn't find which table the original one was a part of, but I'll keep looking!

Reply:
I'm thinking of the `ownership_type` Categorical column.

Reply:
Ack no I'm thinking of `entity_type` here. What do the values in this column look like?  Are they short codes like `I` and `Q` and `F`?  Are they multi-word descriptions like
```py
['Municipal',
 'Cooperative',
 'Political Subdivision',
 'Investor Owned',
 'Power Marketer',
 'Facility',
 'Federal',
 'State',
 'Retail Power Marketer',
 'Unregulated',
 'Wholesale Power Marketer',
 'Behind the Meter',
 'Community Choice Aggregator']
```
Or do they map to those descriptions based on the EIA 861 instructions / documentation?

Reply:
They are both. Some tables (ex: Frames) have an `ownership_code` column AND an `ownership` column fully written out (like `entitiy_type` above).  I see an `entity_type` in the constants `column_dtype` dictionary...should those be the same thing? Neither was categorical until I made `ownership` so with this last commit. The difference between the `ownership` and `ownership_code` columns is that one is a single letter and the other is fully spelled out, but there is a direct correlation between the two.

Reply:
^sorry if that's a bit confusing. happy to elaborate.

Reply:
I think what we want in the long run is to get rid of all the confusing one-letter codes, and replace them with the human readable values like "Wholesale Power Marketer"  It seems like there are a couple of columns in different tables that sometimes have different names, but which are all pointing at the same thing.  There might be some zombie column names in the type dictionary -- things that aren't being used anywhere now because we've consolidated those columns.

Reply:
gotcha -- I'll make a single `ownership` column and add a dictionary that maps the one-letter codes onto the longer version. I've been keeping track of which columns I use so hopefully we'll have a sense of which are zombies in the end.

Reply:
There's already a version of this dictionary in the transform functions for the EIA 860 I think. We should probably pull this out into an EIA level constant so it's only stored in one place.

Reply:
When I tackle the `Frame` table, should I just keep one of the `ownership` columns?

Reply:
Need more context.

Reply:
Agreed. What I did for this table was use the `frame` table as a reference. That seemed to have all of the categories that the `merger` table had and more. Because the `frame` table had both the "code" version and the "full" version of the ownership, I was able to infer what the code in the `merger` table referred to. Admittedly, this is an assumption, but unfortunately the documentation wasn't much help here.

Reply:
I lied - I found a description of the different ownership types that's specific to the `Merger` table. I'll double check it right now.

Reply:
This same column also shows up in the EIA 860 table, and there are different kinds of entities that report to EIA 860 and EIA 861, so all of them may not be represented in either of the forms.  We need to create a consolidated dictionary defining the abbreviations and long names, and in the transform functions, we need to use that dictionary to check whether the abbreviations we're finding match up with our expectations. In the event that we don't have both the abbreviations and the long names available next to each other, we should double check in the form instructions to see that they match up as expected.

Reply:
Ok, I have a dictionary we can work off of in the latest commit. It contains all the code/full name matches from the `Merger` table. There is one more listed in the documentation that does not have an abbreviation associated with it: `Independent Power Producer or Qualifying Facility`. My thoughts are that this corresponds to the `Frame` table's `Unregulated` category, but I need to confirm.

Reply:
When I merge the `Frame` and `Mergers` tables on utility id, the ownership codes match up, so I think it's a safe bet that the full version of the `Frame` ownership also applies to `Mergers`.

Reply:
This is from the documentation: 
<img width="531" alt="Screen Shot 2020-06-29 at 4 34 20 PM" src="https://user-images.githubusercontent.com/49878195/86063770-d39d9700-ba28-11ea-9067-8ea6525eeb96.png">

Reply:
So far as I know the additional types that exist in the EIA 860, which aren't showing up here are just "Industrial" and "Commercial" and @cmgosnell says that's because only "utility" like things respond to EIA 861, while the EIA 860 is filled out by a other kinds of energy producers, even if the energy is getting used on-site.

Reply:
hmm, I'm not sure how we would distinguish "I" codes from `Investor Owned` and `Industrial` as well as `Commerical` from `Cooperative`.

Reply:
In the data itself I'd feel confident running a map match like I did with the `Mergers` and `Frame` tables to confirm the full ownership type. But in the dictionary I don't think we can have an ID (like "I" or "C" represent multiple values)...

Reply:
Luckily, they use `IND` and `COM` to represent industrial and commercial in the EIA 860. There's a dictionary in there already in `pudl.transform.eia860.utilities()` which contains these values as well as others that duplicate the some of the ones from EIA 861.

Reply:
Woo! Thank you EIA -- so two more things: 1) do you think this dictionary should live in constants? I can make a reference in the transform function for 860 and 861 if so. 2) what should the official name of the field be? our options seem to be `ownership`, `ownership_type`, and `entity_type`.

Reply:
Whew, yeah I was afraid there for a moment after your last message that they had really used the same abbreviations to mean different things!  The fact that they didn't makes me feel even better about using the same set of defined categories for these columns. For now I think the dictionary should live in `pudl.constants` -- when we re-organize the column and table definition metadata, we'll pull it (and a bajillion other things) out and organize it alongside a bunch of EIA specific stuff.

`ownership` seems too generic, like it could refer to the fraction of ownership that a particular utility has of a plant or something. To me it feels like this is really a reference to the type of company / organization / entity that's behind the activity and infrastructure that's being referenced.  I guess we've already used `entity` in another context (e.g. utility, plant, balancing authority, generator) so using it here in a different way might be confusing?  We've also used `ownership` in the context of the `ownership_eia860` table (which is also a good example of how we're dealing with table specific `address` etc. columns, check out `pudl_out.own_eia860()`)

But then in the EIA 860 instructions they do use `entity_type` explicitly:

![image](https://user-images.githubusercontent.com/596279/86070097-677b5d00-ba41-11ea-8cce-0ca31ec4a139.png)

While in the EIA 861 they say "Type of Ownership and Function." I'm inclined to go with `entity_type` since some of these descriptive categories don't seem particularly related to the "ownership" aspect and are more about the function of the organization.

Reply:
That all makes sense to me! I'll change it up in 861. Let me know if you'd also like me to add changes to 860.

Reply:
Yes I'd say go ahead and make the one dictionary of abbreviations to categories that can be used for both 860 and 861 and go ahead and refer to it rather than the hard-coded dictionary that's currently being used in 860, and make sure they're both using the same column name for this.